tasks.c

```diff
@@ -1532,9 +1532,10 @@ static void prvInitialiseNewTask( TaskFunction_t pxTaskCode,
     #else /* portSTACK_GROWTH */
     {
         pxTopOfStack = pxNewTCB->pxStack;
+        pxTopOfStack = ( StackType_t * ) ( ( ( ( portPOINTER_SIZE_TYPE ) pxTopOfStack ) + portBYTE_ALIGNMENT_MASK ) & ( ~( ( portPOINTER_SIZE_TYPE ) portBYTE_ALIGNMENT_MASK ) ) ); /*lint !e923 !e9033 !e9078 MISRA exception.  Avoiding casts between pointers and integers is not practical.  Size differences accounted for using portPOINTER_SIZE_TYPE type.  Checked by assert(). */
```

Comment:
Hey @ivq, thanks for raising this PR and being interested in contributing to the FreeRTOS-Kernel!
This change to add the portBYTE_ALIGNMENT_MASK to the memory location the current pxTopOfStack doesn't seem like it would fix the potential issue you are bringing up in the issue you raised #753? 
The increase of the stack size by portBYTE_ALIGNMENT_MASK for this `configASSERT()` would not actually modify the top of the stack as it is used by the system. Where looking at a value of [portBYTE_ALIGNMENT_MASK](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/portable/GCC/ARM_CM33/non_secure/portmacrocommon.h#L97), which in the Cortex-M33 Port is 8, I don't see how this check is any different than the existing one. If pxTopofStack ends in the correct portBYTE_ALIGNMENT_MASK value, simply adding this value to the existing address would lead to a bitwise operation that would return a success.  Where a failure on the existing value would have already returned a success.
Would you mind expanding on the exact issue you are attempting to solve here and potentially provide an example of a possible error case this is attempting to resolve?

Reply:
Suppose that pxNewTCB->pxStack is 0x1000_0004 here and that portBYTE_ALIGNMENT_MASK is 7(portBYTE_ALIGNMENT  is 8). Stack pointer is not properly aligned and the assert in line https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/tasks.c#L1537 will fail.
pxNewTCB->pxStack call be only 4 bytes aligned here if the task is created with xTaskCreateStatic().
See https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/tasks.c#L1198: pxStack is only guaranteed to be aligned as StackType_t, which can be some value smaller than 8(e.g. 4 for 32 bit target).

Reply:
> Suppose that pxNewTCB->pxStack is 0x1000_0004 here and that portBYTE_ALIGNMENT_MASK is 7(portBYTE_ALIGNMENT is 8). 

This assumption itself is not correct. Please see my comment here - https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/753